### PR TITLE
Added Character Replacement Problem

### LINKED ISSUE #1540: Add Character Replacement Problem in sliding window
### Idea Title

Character Replacement Problem

### Idea Description

**Description:**
The Character Replacement problem involves finding the length of the longest substring that can be formed by replacing at most K characters in a given string. This problem is a classic example of the sliding window technique, where we aim to maximize the length of the substring while adhering to the constraint of replacing no more than K characters.

**Problem Statement:** Given a string s and an integer K, determine the length of the longest substring that contains at most K characters that can be replaced. The goal is to find a contiguous substring where the number of characters that are not the most frequent character in that substring is less than or equal to K.

### Potential Benefits

- [X] Improves user experience
- [X] Enhances learning
- [X] Adds a new feature
- [X] Improves code quality
- [ ] Other

@ajay-dhangar I would to work on it/
Please assign it to me under gssoc-ext and hacktoberfest.

### PR DESCRIPTION
**Pull request by tatheer-fathima**
### Description
This pull request includes a solution to the character replacement problem using a sliding window approach. The main objective is to find the longest substring that can be made of the same character by replacing up to k characters in the input string. This approach maximizes the substring length while efficiently handling character replacements.

Fixes #1540 

### Type of change
- [x] New feature (non-breaking change which adds functionality)

### Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

@ajay-dhangar Please review and merge my code under gssoc-ext and hacktoberfest.

### CODE REVIEW

File: docs/Sliding-Window/character-replacement.md [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+int main() {
+    string s = "AABABBA";
+    int k = 1;
+
+    int result = characterReplacement(s, k);
+    cout << "The length of the longest substring is: " << result << endl;
+
+    return 0;
+}
```
Comment (ajay-dhangar) [inline docs/Sliding-Window/character-replacement.md:81]: close your example code with ```` ``` ````

**Review by ajay-dhangar — CHANGES_REQUESTED** [review]
update your code based on suggestions

**Review by tatheer-fathima — COMMENTED** [review]
Ok done.
Made change accordingly..

**Review by ajay-dhangar — APPROVED** [review]
Done

**Pull request merged**